src/components/IncrementStepper/IncrementStepper.jsx

```diff
@@ -0,0 +1,72 @@
+import { React, useCallback, useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import { HiPlusSm, HiMinusSm } from "react-icons/hi";
+
+export const IncrementStepper = ({
+  min,
+  max,
+  initialValue,
+  label,
+  onClick,
+  ...props
+}) => {
+  const [value, setValue] = useState(initialValue);
+  console.log("rerendering component");
+  console.log("current value is: ", value);
+
+  const onIncrease = useCallback(() => {
+    setValue(Math.min(value + 1, max));
+  }, [value]);
+
+  const onDecrease = useCallback(() => {
+    setValue(Math.max(value - 1, min));
+  }, [value]);
+
+  useEffect(() => {
```

Comment:
Funky Way! Wirklich notwendig ist das aber nicht. Könntest ja in den Handlern useClick feuern. :)

Reply:
Dachte ich auch zuerst, aber dann feuert der onChange ja, bevor der value geupdatet ist und der mitgegebene wert wäre noch der alte.